tests/gold_tests/proxy_protocol/proxy_serve_stale_dns_fail.test.py

```diff
@@ -0,0 +1,79 @@
+'''
+Test proxy serving stale content when DNS lookup fails
+'''
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+Test.ContinueOnFail = True
+# Set up hierarchical caching processes
+ts_child = Test.MakeATSProcess("ts_child")
+ts_parent = Test.MakeATSProcess("ts_parent")
+server_name = "http://unknown.domain.com/"
+
+Test.testName = "STALE"
+
+# Config child proxy to route to parent proxy
+ts_child.Disk.records_config.update({
+    'proxy.config.url_remap.pristine_host_hdr': 1,
+    'proxy.config.http.cache.max_stale_age': 10,
+    'proxy.config.http.parent_proxy.self_detect': 0,
+})
+ts_child.Disk.parent_config.AddLine(
+    f'dest_domain=. parent=localhost:{ts_parent.Variables.port} round_robin=consistent_hash go_direct=false'
```

Comment:
So even with this configured, child TS will still do a DNS lookup on unkown.domain.com ?  Did you see the DNS lookup in the http debug trace output?

Reply:
The http debug trace output shows the child doing PPDNS lookup to the parent and the parent doing a DNS lookup to unknown.domain.com (OS). After the DNS lookup fails for the parent and the parent sends a 500 response to the child, the child will first check if the stale object is returnable before sending it back to the client. Let me know if you are seeing something different in the debug trace output!

Reply:
Lately we've been including DNS servers in our Au tests:  https://github.com/apache/trafficserver/blob/57749218b297849e26a00ba80ed0f26358190929/tests/gold_tests/next_hop/strategies_ch/strategies_ch.test.py#L71

Reply:
I don't understand why ATS would do a DNS lookup on the request host when it's not using the request host as the next hop.  But I suppose that's beyond the scope of this PR.